aimlutils/echo/README.md

```diff
@@ -25,8 +25,8 @@ There are three files that must be supplied to use the optimize script:
 The custom **Objective** class (objective.py) must be composed with a **BaseObjective** class (which lives in base_objective.py), and must contain a method named **train** that returns the value of the optimization metric (in a dictionary, see below). There are example objective scripts for both torch and Keras in the examples directory. Your custom Objective class will inherit all of the methods and attributes from the BaseObjective (this way of doing of OOP is called composition). The Objective's train does not depend on the machine learning library used! For example, a simple template has the following structure:
```

Comment:
You are right about inheritance v. composition, that it's really more of a mixture, thanks for the spot! 

Back when I first wrote the base objective, I wanted it to do three things: hand off the device ID(s) (for torch models, and for multi-gpu support later on), update as many fields of the model config automatically, and then do a structured save, where the user could have this method save whatever was in the results dictionary. The later part pre-dated report.py, so that's mostly already been cut. Long story short, it doesn't really do that much. I have a draft version that's working with just a custom, user-supplied method, rather than class (which is what optuna originally intended).